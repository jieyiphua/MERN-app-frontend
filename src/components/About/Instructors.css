
.instructors-container {
  display: flex;
  flex-wrap: wrap;
  flex-direction: space-around;
  justify-content: center;
  align-items: center;
  width: 100%;
}





.instructor-detail-content h2 {
  border-bottom: 1px solid black;
}



.instructor-detail-main {
  height: 300px;
  width: 40%;
  margin: 20px;
  padding: 10px 10px;
  background-color: white;
  /* border: 1pt solid #e1e1e1; */
  border: 1px solid black;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.instructor-detail-content {
  width: 90%;
  height: 100%;
  margin-top: 10px;
  margin-bottom: 15px;
  display: flex;
  justify-content: space-between;
  align-items: space-evenly;
  flex-direction: column;
}


.instructors-delete-button {
  text-decoration: none;
  background: none;
  border: 1px solid red;
  padding: 10px;
  margin: 2px;
  background-color: white;
  color: red;
  display: block;
  width: 150px;

}

.instructors-update-button {
  text-decoration: none;
  background: none;
  border: 1px solid black;
  padding: 10px;
  margin: 2px;
  background-color: white;
  display: block;
  width: 150px;
}

