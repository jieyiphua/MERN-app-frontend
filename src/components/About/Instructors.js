import React from "react";
import "./Instructors.css";
import axios from "axios";
import { Link, withRouter } from "react-router-dom";

class Instructors extends React.Component {
  state = {
    data: [],
    errors: []
  };

  async componentDidMount() {
    this.getInstructors();
  }

  getInstructors = async () => {
    const response = await axios.get(
      "https://deployment-mern-backend-tessivanjayz.gardtess.now.sh/instructor"
    );
    console.log(response);
    this.setState({
      data: response.data
    });
  };

  deleteInstructor = async item => {
    console.log(item);
    try {
      const response = await axios.delete(
        `https://deployment-mern-backend-tessivanjayz.gardtess.now.sh/instructor/${
          item._id
        }`
      );
      console.log(response);
      this.getInstructors();
    } catch (error) {
      this.setState({ errors: error.response });
    }
  };

  render() {
    const { onContentSelect, authentication } = this.props;
    console.log("render");
    if (authentication) {
      return (
        <div className="main-container">
          <div className="inner-main-container">
            <div className="new-class-header" id="about-header">
              <h1>Faculty</h1>
            </div>
            <Link to="/admin/instructor/new">
              <button id="new-about-button" className="class-button">
                Add New Instructor
              </button>
            </Link>

            <div className="instructors-container">
              {this.state.data.map((item, index) => (
                <div className="instructor-detail-main" key={index}>
                  <div className="instructor-detail-content">
                    <h2>{item.name}</h2>
                    <h4>Expertise: {item.description}</h4>
                    <h4>Faculty : {item.faculty}</h4>
                    <div>
                      <Link
                        to="/admin/instructor/update"
                        onClick={() => onContentSelect(item)}
                      >
                        <button className="class-button">Update Details</button>
                      </Link>
                      <button
                        id="instructor-delete"
                        className="class-button"
                        onClick={() => this.deleteInstructor(item)}
                      >
                        Delete Instructor
                      </button>
                    </div>
                  </div>
                </div>
              ))}
            </div>
          </div>
        </div>
      );
    } else {
      return (
        <div className="main-container">
          <div className="inner-main-container">
            <div className="new-class-header" id="about-header">
              <h1>Faculty</h1>
            </div>
            <br />
            <div className="instructors-container">
              {this.state.data.map((item, index) => (
                <div className="instructor-detail-main" key={index}>
                  <div className="instructor-detail-content">
                    <h2>{item.name}</h2>
                    <h4>Expertise: {item.description}</h4>
                    <h4>Faculty : {item.faculty}</h4>
                  </div>
                </div>
              ))}
            </div>
          </div>
        </div>
      );
    }
  }
}
export default withRouter(Instructors);
