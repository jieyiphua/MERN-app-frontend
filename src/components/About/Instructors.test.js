import Instructors from "./Instructors";
import React from "react";
import Enzyme, { shallow, render, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { createSerializer } from "enzyme-to-json";
import sinon from "sinon";

describe("<Instructors />", () => {
  it("renders component", () => {
    const wrapper = shallow(<Instructors />);
    expect(wrapper).toMatchSnapshot();
  });
});
