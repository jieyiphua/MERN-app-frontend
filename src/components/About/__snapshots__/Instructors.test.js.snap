// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Instructors /> renders component 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
